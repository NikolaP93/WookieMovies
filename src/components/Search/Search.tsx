import React, {useState, useEffect} from 'react';
import {
  StyleSheet,
  TextInput,
  Dimensions,
  View,
  ScrollView,
  TouchableOpacity,
  KeyboardAvoidingView,
} from 'react-native';
import constants from '../../constants';
import useMoviesData from '../../modules/hooks/useMoviesData';
import debounce from 'lodash.debounce';
import Text from '../Text/Text';
import Movie from '../Movie/Movie';
import {useNavigation} from '@react-navigation/native';
import {HomeRoutes} from '../../navigation/config/Routes';

const {height, width} = Dimensions.get('screen');

const SearchInput = () => {
  const [query, setQuery] = useState('');
  const {queryMovies, queriedMovies} = useMoviesData();
  const navigation = useNavigation();

  const debouncedQuery = debounce(queryMovies, 800);

  useEffect(() => {
    debouncedQuery(query);
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [query]);

  const onPress = (movieTitle: string) => {
    navigation.navigate(HomeRoutes.details, {movieTitle});
  };

  return (
    <View style={styles.mainContainer}>
      <TextInput style={styles.container} onChangeText={setQuery} />
      {queriedMovies.length > 0 && (
        <ScrollView style={styles.scrollView}>
          {queriedMovies.map(movie => (
            <TouchableOpacity
              key={movie.id}
              style={styles.results}
              onPress={() => onPress(movie.title)}>
              <Movie uri={movie.poster} />
              <Text>{movie.title}</Text>
            </TouchableOpacity>
          ))}
        </ScrollView>
      )}
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    height: height / 15,
    width: '100%',
    borderWidth: 1,
    backgroundColor: constants.colors.white,
    borderRadius: 10,
    paddingHorizontal: 15,
    marginVertical: 10,
  },
  mainContainer: {
    paddingHorizontal: 10,
    alignItems: 'center',
    width: (width * 4) / 5,
  },
  scrollView: {
    flexGrow: 1,
    width: '100%',
  },
  results: {
    flexDirection: 'row',
    height: 100,
    alignItems: 'center',
    width: '100%',
    marginVertical: 1,
    borderWidth: 1,
    paddingVertical: 5,
  },
});

export default SearchInput;
