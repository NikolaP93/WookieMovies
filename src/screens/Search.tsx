import React from 'react';
import {StyleSheet, View} from 'react-native';
import {SafeAreaView} from 'react-native-safe-area-context';
import SearchInput from '../components/Search/Search';
import constants from '../constants';

const Search = (): JSX.Element => {
  return (
    <SafeAreaView style={styles.container}>
      <View style={styles.searchContainer}>
        <SearchInput />
      </View>
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    backgroundColor: constants.colors.white,
  },
  searchContainer: {
    marginVertical: 30,
    flex: 1,
  },
});

export default Search;
