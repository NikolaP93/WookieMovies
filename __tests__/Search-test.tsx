import 'react-native';
import React from 'react';
import Search from '../src/screens/Search';
import {create} from 'react-test-renderer';
jest.useFakeTimers();
jest.mock('react-native-gesture-handler', () => {});
jest.mock('@react-navigation/native');

const tree = create(<Search />);

test('takes snapshot of search screen', () => {
  expect(tree).toMatchSnapshot();
});
