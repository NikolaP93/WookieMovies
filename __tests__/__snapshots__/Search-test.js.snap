// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`takes snapshot of search screen 1`] = `
<RNCSafeAreaView
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "marginVertical": 30,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "paddingHorizontal": 10,
          "width": 600,
        }
      }
    >
      <TextInput
        onChangeText={[Function]}
        style={
          Object {
            "backgroundColor": "white",
            "borderRadius": 10,
            "borderWidth": 1,
            "height": 88.93333333333334,
            "marginVertical": 10,
            "paddingHorizontal": 15,
            "width": "100%",
          }
        }
      />
    </View>
  </View>
</RNCSafeAreaView>
`;
